src/Symfony/Component/Process/Process.php

```diff
@@ -196,7 +197,11 @@ public static function fromShellCommandline(string $command, string $cwd = null,
 
     public function __destruct()
     {
-        $this->stop(0);
+        if (!$this->options['create_new_console']) {
```

Comment:
`If (!(this->options['detach'] ?? $this->options['create_new_console']))`

Reply:
Do you think  is a good idea to have the existence of `this->options['detach']`? 
because `$this->options` is passed to `proc_open` in `start()` and obviously `detach` is not a valid `proc_open` flag.

My though was that is was better when using setOption(['detach'] => true) to just set the $option['create_new_console'] according to `detach`.

 Also the suggested check is misleading since on Windows `CREATE_NEW_CONSOLE` flag always detaches the process, that's why `CREATE_NEW_CONSOLE` is incompatible with the flag `DETACHED`(not supported by proc_open) in `CreateProcess`, the difference between `DETACHED` and  `CREATE_NEW_CONSOLE` is that the  `CREATE_NEW_CONSOLE` simply attaches a new console to the detached process whereas in the case of `DETACHED` you can attach a console later. 

I can change as suggested but then I think the call of `proc_open` needs to be changed so when it's called the $option->detach is omitted.

Reply:
> My though was that is was better when using setOption(['detach'] => true) to just set the $option['create_new_console'] according to detach.

I see it the other way around: when `create_new_console` is set, our special `detach` option should be set to true on both Windows and Linux. But it one wants to pass `create_new_console=true + detach=false`, it should be allowed to.

Note that the "detach" you mention is not the same as the one we're talking about here: here, we're talking about detaching from the current process. Microsoft's detach is about detaching from a display console (and is not accessible from PHP you're right).

> detach is not a valid proc_open flag.

yes, but proc_open silently accepts it, isn't it? then no need to care imho. If MS' option is supported one day, it will be named `detach_process`

Maybe we can find another name for our pseudo option? Alternatively, we could use a new `setDetached()` method to configure the behavior?

Btw this made me think: setOptions() should start with 
```php
        if ($this->isRunning()) {
            throw new RuntimeException('Setting options while the process is running is not possible.');
        }
```

Reply:
I made the changes, I agree, that a marginally case can be to wait for the detached process `(create_new_console=true + detach=false)`, although I think `create_new_console=false + detach=true` on Windows should not be legal since without `create_new_console` you can't terminate your main script and let the process run, you can't even start the process in detached mode, mostly I guess is not a big problem if in documentation of the Process component that is specified.